packages/renderer/server/main.ts

```diff
@@ -49,10 +50,27 @@ async function renderWorker(
     }
   });
 
+  const workerLines: Record<string, number> = {};
+  page.exposeFunction('logProgress', (port: string, progress: number) => {
```

Comment:
I don't think we have to send the port here, we can just use the port that is defined further up outside of the callback.